diff --git a/Makefile b/Makefile
index 0fb1f84..1fb8b03 100644
--- a/Makefile
+++ b/Makefile
@@ -58,8 +58,9 @@ install-manpages: $(MANPAGES)
 
 install: install-mnexec install-manpages
 #	This seems to work on all pip versions
-	$(PYTHON) -m pip uninstall -y mininet || true
-	$(PYTHON) -m pip install .
+# 	$(PYTHON) -m pip uninstall -y mininet || true
+# 	$(PYTHON) -m pip install .
+	python3 setup.py install
 
 develop: $(MNEXEC) $(MANPAGES)
 # 	Perhaps we should link these as well
diff --git a/util/install.sh b/util/install.sh
index bf4492d..6187a0d 100755
--- a/util/install.sh
+++ b/util/install.sh
@@ -171,39 +171,28 @@ function mn_deps {
 		$install gcc make socat psmisc xterm openssh iperf \
 			iproute telnet ${PYPKG}-setuptools libcgroup-tools \
 			ethtool help2man python-pyflakes python3-pylint \
-                        python-pep8 ${PYPKG}-pexpect ${PYPKG}-tk
+                        python-pycodestyle ${PYPKG}-pexpect ${PYPKG}-tk
     else  # Debian/Ubuntu
-        pf=pyflakes
-        pep8=pep8
-        # Starting around 20.04, installing pyflakes instead of pyflakes3
-        # causes Python 2 to be installed, which is exactly NOT what we want.
-        if [ "$DIST" = "Ubuntu" -a `expr $RELEASE '>=' 20.04` = "1" ]; then
-                pf=pyflakes3
-        fi
-        # Debian 11 "bullseye" renamed
-        # * pep8 to python3-pep8
-        # * pyflakes to pyflakes3
-        if [ "$DIST" = "Debian" -a `expr $RELEASE '>=' 11` = "1" ]; then
-                pf=pyflakes3
-                pep8=python3-pep8
-        fi
+        pf=python3-pyflakes
+        pep8=python3-pycodestyle
 
         $install gcc make socat psmisc xterm ssh iperf telnet \
                  ethtool help2man $pf pylint $pep8 \
                  net-tools ${PYPKG}-tk
 
         # Install pip
-        $install ${PYPKG}-pip || $install ${PYPKG}-pip-whl
-        if ! ${PYTHON} -m pip -V; then
-            if [ $PYTHON_VERSION == 2 ]; then
-                wget https://bootstrap.pypa.io/pip/2.7/get-pip.py
-            else
-                wget https://bootstrap.pypa.io/get-pip.py
-            fi
-            sudo ${PYTHON} get-pip.py
-            rm get-pip.py
-        fi
-       ${python} -m pip install pexpect
+        # $install ${PYPKG}-pip || $install ${PYPKG}-pip-whl
+        # if ! ${PYTHON} -m pip -V; then
+        #     if [ $PYTHON_VERSION == 2 ]; then
+        #         wget https://bootstrap.pypa.io/pip/2.7/get-pip.py
+        #     else
+        #         wget https://bootstrap.pypa.io/get-pip.py
+        #     fi
+        #     sudo ${PYTHON} get-pip.py
+        #     rm get-pip.py
+        # fi
+    #    ${python} -m pip install pexpect
+        $install python3-pexpect
         $install iproute2 || $install iproute
         $install cgroup-tools || $install cgroup-bin
         $install cgroupfs-mount
@@ -222,7 +211,8 @@ function mn_doc {
     if ! $install doxygen-latex; then
         echo "doxygen-latex not needed"
     fi
-    sudo pip2 install doxypy
+    $install python-doxypy
+    # sudo pip2 install doxypy
 }
 
 # The following will cause a full OF install, covering:
